be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -1209,6 +1291,53 @@ Status SegmentIterator::next_batch(vectorized::Block* block) {
     return Status::OK();
 }
 
+void SegmentIterator::_output_index_return_column(uint16_t* sel_rowid_idx, uint16_t select_size, vectorized::Block* block) {
+    if (block->rows() == 0) {
+        return;
+    }
+
+    for (auto column_sign : _rowid_result_for_index) {
+        block->insert({vectorized::ColumnUInt8::create(),
+                       std::make_shared<vectorized::DataTypeUInt8>(), column_sign.first});
+        if (!column_sign.second.first) {
+            // predicate not in compound query
+            block->get_by_name(column_sign.first).column =
+                    vectorized::DataTypeUInt8().create_column_const(block->rows(), 1u);
+            continue;
+        }
+        _build_index_return_column(sel_rowid_idx, select_size, block, column_sign.first, column_sign.second.second);
+    }
+}
+
+void SegmentIterator::_build_index_return_column(uint16_t* sel_rowid_idx, uint16_t select_size,
```

Comment:
add comment for the logic for row range and id split.

Reply:
updated